wordpress-mobile/WordPress-Android
# Contact info api - featureConfig implementation

This PR builds on top of https://github.com/wordpress-mobile/WordPress-Android/pull/13758 and implements the FeatureConfig class.

Related Gutenberg-mobile: PR https://github.com/wordpress-mobile/gutenberg-mobile/pull/3015 and https://github.com/wordpress-mobile/gutenberg-mobile/pull/3001
Related Gutenberg PR: https://github.com/WordPress/gutenberg/pull/28293


**To test:**

CASE A: flag ON for production
1. build vanilla with this PR as is
2. on a jetpack site with version 8.5 and above, create a GB Post
3. verify the block picker in mobile gutenberg shows the Contact Info block and you can insert and use it in the Post.
4. switch the site to a self hosted site, or a site using a jetpack version older than 8.5
5. verify the block picker in mobile gutenberg does NOT show the Contact Info block

CASE B: flag OFF for production
1. in `WordPress/build.gradle` edit this line in the vanilla flavor `buildConfigField "boolean", "CONTACT_INFO_BLOCK_AVAILABLE", "true"` and set it ti `false` instead
2. build vanilla
3. on a jetpack site with version 8.5 and above, create a GB Post
4. verify the block picker in mobile gutenberg does NOT show the Contact Info block

PR submission checklist:
